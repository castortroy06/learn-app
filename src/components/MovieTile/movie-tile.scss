@import '../../sass/bootstrap/variables';

.movie-tile {
    position: relative;
    background-color: $gray-300;
    padding-bottom: 1.25rem;
    margin-bottom: 1.9375rem;

    > a {
        text-decoration: none;
        color: inherit;
    }

    img {
        margin-bottom: 1.375rem;
        display: block;
        width: 100%;
    }

    &--container {
        display: flex;
        justify-content: space-between;
    }

    &--name {
        font-size: 1.125rem;
        font-weight: 500;
        line-height: 1.2188;
        color: $white;
        margin-bottom: 0.5rem;
    }

    &--year,
    &--categories {
        font-size: 0.875rem;
        font-weight: 500;
        line-height: 1.2193;
        color: $white;
    }

    &--categories {
        opacity: 0.5;
    }

    &--year {
        border: 1px solid $gray-500;
        padding: 0.25rem 0.5rem;
        margin-bottom: 0.5rem;
        border-radius: 0.25rem;
        opacity: 0.5;
    }
}
