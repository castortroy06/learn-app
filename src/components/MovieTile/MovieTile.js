import React from "react";
import './movie-tile.scss';

const MovieTile = ({ imageUrl, name, releaseYear, genres, onClick }) => {
    return (
        <div className="movie-tile col-4" onClick={onClick}>
            <img data-testid="image" alt={name} src={`/images/movies/${imageUrl}`} />

            <div className="movie-tile--container">
                <h6 data-testid="name" className="movie-tile--name display-6">{name}</h6>
                <p data-testid="year" className="movie-tile--year col-15">{releaseYear}</p>
            </div>

            <p data-testid="categories" className="movie-tile--categories">{genres.join(', ')}</p>
        </div>
    );
};

export default MovieTile;
