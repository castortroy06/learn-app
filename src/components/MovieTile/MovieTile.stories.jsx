import MovieTile from './MovieTile';

export default {
    title: 'Movie Tile',
}

const handleClick = () => {
    
};

const movieTile = {
    'name': 'Pulp Fiction',
    'imageUrl': 'pulp-fiction.png',
    'releaseYear': 2004, 
    'genres': ['Action & Adventure'],
    'onClick': handleClick(),
}

export const Default = () => <MovieTile {...movieTile} />