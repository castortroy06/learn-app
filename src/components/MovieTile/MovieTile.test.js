import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import MovieTile from './MovieTile';

describe('MovieTile', () => {
    it('match snapshot', () => {
        const handleClick = () => {
            
        };

        const options = {
            'name': 'Pulp Fiction',
            'imageUrl': 'pulp-fiction.png',
            'releaseYear': 2004,
            'genres': ['Action & Adventure'],
            'onClick': handleClick(),
        }

        const tree = renderer.create(<MovieTile {...options} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
