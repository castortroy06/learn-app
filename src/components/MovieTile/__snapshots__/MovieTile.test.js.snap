// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTile match snapshot 1`] = `
<div
  className="movie-tile col-4"
>
  <img
    alt="Pulp Fiction"
    data-testid="image"
    src="/images/movies/pulp-fiction.png"
  />
  <div
    className="movie-tile--container"
  >
    <h6
      className="movie-tile--name display-6"
      data-testid="name"
    >
      Pulp Fiction
    </h6>
    <p
      className="movie-tile--year col-15"
      data-testid="year"
    >
      2004
    </p>
  </div>
  <p
    className="movie-tile--categories"
    data-testid="categories"
  >
    Action & Adventure
  </p>
</div>
`;

exports[`MovieTile renders all elements successfully 1`] = `
<div
  className="movie-tile col-4"
>
  <img
    alt="Pulp Fiction"
    data-testid="image"
    src="/images/movies/pulp-fiction.png"
  />
  <div
    className="movie-tile--container"
  >
    <h6
      className="movie-tile--name display-6"
      data-testid="name"
    >
      Pulp Fiction
    </h6>
    <p
      className="movie-tile--year col-15"
      data-testid="year"
    >
      2004
    </p>
  </div>
  <p
    className="movie-tile--categories"
    data-testid="categories"
  >
    Action & Adventure
  </p>
</div>
`;
